packages/core/echo/echo-pipeline/src/pipeline/pipeline.ts

```diff
@@ -372,11 +372,15 @@ export class Pipeline implements PipelineAccessor {
 
   private _setFeedDownloadState(feed: FeedWrapper<FeedMessage>) {
     const timeframe = this._state._startTimeframe;
-    const seq = timeframe.get(feed.key) ?? 0;
-
-    feed.undownload({ callback: () => log('undownload') });
-    feed.download({ start: seq + 1, linear: true }).catch((err: Error) => {
-      log('failed to download feed', { err });
+    const seq = timeframe.get(feed.key) ?? -1;
+
+    feed.undownload({
```

Comment:
hmm, is this better then calling them in parallel? I'm concerned about possible race conditions if _setFeedDownloadState is called again before undownload callback could complete

Reply:
ok good point